cinema/views.py

```diff
@@ -98,11 +101,28 @@ def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
 
-        if self.action == "retrieve":
+        elif self.action == "retrieve":
             return MovieDetailSerializer
 
+        elif self.action == "upload_image":
+            return MovieImageSerializer
```

Comment:
The `get_serializer_class` method in `MovieViewSet` correctly returns different serializers based on the action. Ensure that the `upload_image` action is properly defined in the router to match this method.